pydra/engine/core.py

```diff
@@ -700,6 +700,7 @@ def add(self, task):
             raise Exception("Submitter should already be set.")
         # at this point Workflow is stateless so this should be fine
         nwf = await submitter.submit(self, return_task=True)
+        # TODO: check if needed
```

Comment:
is this still needed?

Reply:
are you asking if my note is needed? I left it to remember about thinking about it, can rephrase